packages/e2e-tests/specs/autosave.test.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	clickBlockAppender,
+	createNewPost,
+	getEditedPostContent,
+	pressKeyWithModifier,
+} from '@wordpress/e2e-test-utils';
+
+// Constant to override editor preference
+const AUTOSAVE_INTERVAL_SECONDS = 5;
+
+async function saveDraftWithKeyboard() {
+	return pressKeyWithModifier( 'primary', 's' );
+}
+
+async function sleep( durationInSeconds ) {
+	return new Promise( ( resolve ) =>
+		setTimeout( resolve, durationInSeconds * 1000 ) );
+}
+
+async function clearSessionStorage() {
+	await page.evaluate( () => window.sessionStorage.clear() );
+}
+
+async function readSessionStorageAutosave( postId ) {
+	return page.evaluate(
+		( key ) => window.sessionStorage.getItem( key ),
+		`wp-autosave-block-editor-post-${ postId }`
+	);
+}
+
+async function getCurrentPostId() {
+	return page.evaluate(
+		() => window.wp.data.select( 'core/editor' ).getCurrentPostId()
+	);
+}
+
+async function setLocalAutosaveInterval( value ) {
+	return page.evaluate( ( _value ) => {
+		window.wp.data.dispatch( 'core/edit-post' )
+			.__experimentalUpdateLocalAutosaveInterval( _value );
+	}, value );
+}
+
+function wrapParagraph( text ) {
+	return `<!-- wp:paragraph -->
+<p>${ text }</p>
+<!-- /wp:paragraph -->`;
+}
+
+describe( 'autosave', () => {
+	beforeEach( async () => {
+		await clearSessionStorage();
+		await createNewPost();
+		await setLocalAutosaveInterval( AUTOSAVE_INTERVAL_SECONDS );
```

Comment:
Should there be an associated tear down to this, to avoid the state bleeding over into other tests run in the same container?

Not sure it would have any noticeable impact since ideally those tests would be resilient regardless of an autosave interval, but seems like an undesirable side-effect in any case.

Reply:
I guess it might depend if `setLocalAutosaveInterval` is something which would persist outside the current page. If most tests would create a new post and thus navigate away, it might not be a problem.

Then again, it's long been a desire of mine that `createNewPost` shouldn't have to trigger a full page reload (see #18668).